vcr/issuer/leia_store_test.go

```diff
@@ -376,6 +376,43 @@ func TestLeiaIssuerStore_handleRestore(t *testing.T) {
 	})
 }
 
+func TestNewLeiaIssuerStore(t *testing.T) {
+	t.Run("bug test for https://github.com/nuts-foundation/nuts-node/issues/1909", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		mockStore := stoabs.NewMockKVStore(ctrl)
+		backupStorePath := path.Join(t.TempDir(), "backup-issued-credentials.db")
+		emptyBackupStore, err := bbolt.CreateBBoltStore(backupStorePath)
+		dbPath := path.Join(t.TempDir(), "issuer.db")
+
+		// first create a store with 1 credential
+		store, err := NewLeiaIssuerStore(dbPath, emptyBackupStore)
+		require.NoError(t, err)
+		vc := vc.VerifiableCredential{}
+		_ = json.Unmarshal([]byte(jsonld.TestCredential), &vc)
+		require.NoError(t, store.StoreCredential(vc))
+		require.NoError(t, store.Close())
+
+		// now create a new store with a mock backup and show that ReadShelf only called once.
+		// no additional calls
+		reader := stoabs.NewMockReader(ctrl)
+		reader.EXPECT().Iterate(gomock.Any(), gomock.Any()).DoAndReturn(func(argCB interface{}, argT interface{}) error {
```

Comment:
Had to dive into the code to see why this works, it's because `backupStorePresent` calls Iterate and ReadShelf, if restore happens it'll be called again by the restore. But it's quite complicated, so maybe add a comment as to why the mock behavior it is as it is.